s3transfer/utils.py

```diff
@@ -300,7 +300,12 @@ def is_special_file(cls, filename):
         return False
 
     def get_temp_filename(self, filename):
-        return filename + os.extsep + random_file_extension()
+        max_filename_len = 255
+        sep = os.extsep
+        ext = random_file_extension()
+        path,name = os.path.dirname(filename), os.path.basename(filename)
+        temp_filename = name[:max_filename_len-len(sep)-len(ext)] + sep + ext 
```

Comment:
Let's add spaces around the operators like so `max_filename_len - len(sep) - len(ext)`.

If the line starts to get too long it might make sense to break it up and store some of these calculations in a variable and then use them in the array range.